lib/hydra/pcdm/models/concerns/collection_behavior.rb

```diff
@@ -38,6 +38,8 @@ def << arg
 
     def collections= collections
       raise ArgumentError, "each collection must be a Hydra::PCDM::Collection" unless collections.all? { |c| collection? c }
+      raise ArgumentError, "you can't add a collection to itself" if collections.all? { |c| c.id == self.id }
```

Comment:
yeah, "add a collection to itself" is just a special case of "recusively nest collections"